worlds/kh1/Rules.py

```diff
@@ -0,0 +1,1948 @@
+from BaseClasses import CollectionState
+from worlds.generic.Rules import add_rule
+
+SINGLE_PUPPIES = ["Puppy " + str(i).rjust(2,"0") for i in range(1,100)]
+TRIPLE_PUPPIES = ["Puppies " + str(3*(i-1)+1).rjust(2, "0") + "-" + str(3*(i-1)+3).rjust(2, "0") for i in range(1,34)]
+TORN_PAGES = ["Torn Page " + str(i) for i in range(1,6)]
+WORLDS =    ["Wonderland", "Olympus Coliseum", "Deep Jungle", "Agrabah",      "Monstro",      "Atlantica", "Halloween Town", "Neverland",  "Hollow Bastion", "End of the World"]
+KEYBLADES = ["Lady Luck",  "Olympia",          "Jungle King", "Three Wishes", "Wishing Star", "Crabclaw",  "Pumpkinhead",    "Fairy Harp", "Divine Rose",    "Oblivion"]
+
+def has_x_worlds(state: CollectionState, player: int, num_of_worlds: int, keyblades_unlock_chests: bool) -> bool:
+    worlds_acquired = 0.0
+    for i in range(len(WORLDS)):
+        if state.has(WORLDS[i], player):
+            worlds_acquired = worlds_acquired + 0.5
+        if (state.has(WORLDS[i], player) and (not keyblades_unlock_chests or state.has(KEYBLADES[i], player))) or (state.has(WORLDS[i], player) and WORLDS[i] == "Atlantica"):
+            worlds_acquired = worlds_acquired + 0.5
+    return worlds_acquired >= num_of_worlds
+
+def has_emblems(state: CollectionState, player: int, keyblades_unlock_chests: bool) -> bool:
+    return state.has_all({
+        "Emblem Piece (Flame)",
+        "Emblem Piece (Chest)",
+        "Emblem Piece (Statue)",
+        "Emblem Piece (Fountain)",
+        "Hollow Bastion"}, player) and has_x_worlds(state, player, 5, keyblades_unlock_chests)
+
+def has_puppies_all(state: CollectionState, player: int, puppies_required: int) -> bool:
+    return state.has("All Puppies", player)
+
+def has_puppies_triplets(state: CollectionState, player: int, puppies_required: int) -> bool:
+    return state.has_from_list_unique(TRIPLE_PUPPIES, player, -(puppies_required / -3))
```

Comment:
Oh this is a clever idea. However, dividing an int by an int makes a float in Python. If you wanted integer division, use `//` instead of `/`

Reply:
(You can also just do `puppies_required` directly, it's technically fine if you pass a float to that function, although an int is "cleaner")

Reply:
I think the double negative should also be removed, no?

Reply:
Is there a reason this doesn't just use math.ceil?

Reply:
This never got fixed or addressed